File: backend/app/src/models/buildings/Warehouse.py
from ..blueprints.Factory import Factory


class Warehouse(Factory):
    def __init__(self, workers: list = [], 
                    storage_capacity = {
                        "gold": 2000,
                        "food": 2000,
                        "wood": 2000,
                        "iron": 2000
                    },
                    current_stock = {
                        "gold": 0,
                        "food": 0,
                        "wood": 0,
                        "iron": 0
                    }
                 ) -> None:
        super().__init__(name="Warehouse", worker_slots=2, workers=workers)
        self.storage_capacity: dict = storage_capacity
        self.__current_stock: dict = current_stock
    
    
    def get_current_stock(self) -> dict:
        return self.__current_stock
    
    
    def convert_worker_to_wh_worker(self, worker, location) -> bool:
        if self.convert_worker_to_craftsman("Warehouse", worker, location):
            return True

        return False
 







if __name__ == "__main__":
    pass

File: backend_refactoring/CSharpApi/Models/BluePrints/BuildingTypes/Warehouse.cs

using CSharpApi.Models.BluePrints.Beings;
using CSharpApi.Models.BluePrints.Locations;

namespace CSharpApi.Models.BluePrints.BuildingTypes
{
    public class Warehouse : Factory, IBuildingList
    {
        public string Name { get; set; } = "Warehouse";

        private List<Worker> _workers;

        public Dictionary<string, Dictionary<string, float>> Costs { get; set; } = new() {
            { "costs", new()
                {
                    { "gold", 200 },
                    { "food", 200 },
                    { "wood", 300 },
                    { "iron", 100 }
                }
            }
        };

        public readonly Dictionary<string, int> StorageCapacity;

        private Dictionary<string, int> _currentCapacity;

        public Warehouse(List<Worker> workers = null)
           : base("Warehouse", true, 2, workers ?? [])
        {
            _workers ??= workers ?? [];
            StorageCapacity = new() { { "gold", 2000 }, { "food", 2000 }, { "wood", 2000 }, { "iron", 2000 } };
            _currentCapacity = new() { { "gold", 0 }, { "food", 0 }, { "wood", 0 }, { "iron", 0 } };
        }

        public Dictionary<string, int> GetCurrentCapacity() { return _currentCapacity; }

        public bool ConvertWorkerToWarehouseWorker(int workerId, Settlement location)
        {
            return (bool)ConvertWorkerToCraftsman(Name, workerId, location)["success"];
        }
    }
}
